src/t_hash.c

```diff
@@ -107,7 +110,7 @@ sds hashTypeGetFromHashTable(robj *o, sds field) {
  * can always check the function return by checking the return value
  * for C_OK and checking if vll (or vstr) is NULL. */
 int hashTypeGetValue(robj *o, sds field, unsigned char **vstr, unsigned int *vlen, long long *vll) {
-    if (o->encoding == OBJ_ENCODING_ZIPLIST) {
+    if (o->encoding == OBJ_ENCODING_ZIPLIST || o->encoding == OBJ_ENCODING_LISTPACK) {
```

Comment:
maybe create some IS_PACKED macro for that in server.h?
in combination with the PACKED_CLASS macro suggestion, much of the code that works with that doesn't need to ever mention listpack or ziplist, and will remain the same if we ever add a 3rd packed encoding type.

same goes for some portions of code that only need to save work with the pointer or blob, like rdb.c or defrag.c

Reply:
good job!